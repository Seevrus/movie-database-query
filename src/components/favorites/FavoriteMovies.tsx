import { Button, Card, Col, Container, Row } from 'react-bootstrap';
import { Movie } from '../../model/Movie';

interface FavoriteMovieProps {
  favoriteMovies: Movie[];
  focusedMovie?: Movie | undefined;
  setFocusedMovie: React.Dispatch<React.SetStateAction<Movie | undefined>>;
}

const FavoriteMovies = ({ favoriteMovies, focusedMovie, setFocusedMovie }: FavoriteMovieProps) => {
  const handleClick = (m: Movie, isFocused:boolean) => {
    if (isFocused) {
      setFocusedMovie(undefined);
    } else {
      setFocusedMovie(m);
    }
  };

  if (favoriteMovies.length === 0) return <></>;

  return (
    <Container fluid>
      <Row className="justify-content-md-center">
        <Col md="8">
          <Card className="login">
            <Card.Header>
              <Card.Title as="h4">Kedvencek</Card.Title>
            </Card.Header>
            <Card.Body>
                {favoriteMovies.map((movie) => {
                  const isMovieFocused = movie.id === focusedMovie?.id;
                  return (
                    <Button
                      variant={isMovieFocused ? 'primary' : 'light'}
                      key={movie.id}
                      onClick={() => handleClick(movie, isMovieFocused)}
                    >
                      {movie.title}
                    </Button>
                  );
                })}
            </Card.Body>
          </Card>
        </Col>
      </Row>
    </Container>
  );
};

export default FavoriteMovies;
