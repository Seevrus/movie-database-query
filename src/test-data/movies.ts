import { Movie } from '../model/Movie';

const movies = [
  new Movie({
    id: 1,
    title: 'Reality Bites',
  }),
  new Movie({
    id: 2,
    title: 'Demolition Man',
  }),
  new Movie({
    id: 3,
    title: 'Kramer vs. Kramer',
  }),
  new Movie({
    id: 4,
    title: 'The Christmas Chronicles: Part Two',
  }),
  new Movie({
    id: 5,
    title: 'The Ides of March',
  }),
  new Movie({
    id: 6,
    title: 'The Princess Switch 3: Romancing the Star',
  }),
];

export default movies;
