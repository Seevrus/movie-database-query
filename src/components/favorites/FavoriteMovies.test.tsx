import renderer from 'react-test-renderer';
import movies from '../../test-data/movies';
import FavoriteMovies from './FavoriteMovies';

describe('ResultsPagination', () => {
  it('Should render correctly', () => {
    const component = renderer.create(
      <FavoriteMovies favoriteMovies={movies} setFocusedMovie={jest.fn()} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should render correctly if there is a focused movie', () => {
    const component = renderer.create(
      <FavoriteMovies
        favoriteMovies={movies}
        focusedMovie={movies[0]}
        setFocusedMovie={jest.fn()}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
