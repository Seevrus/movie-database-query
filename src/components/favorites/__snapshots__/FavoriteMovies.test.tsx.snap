// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPagination Should render correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-md-8"
    >
      <div
        className="login card"
      >
        <div
          className="card-header"
        >
          <h4
            className="card-title"
          >
            Kedvencek
          </h4>
        </div>
        <div
          className="card-body"
        >
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Reality Bites
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Demolition Man
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Kramer vs. Kramer
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            The Christmas Chronicles: Part Two
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            The Ides of March
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            The Princess Switch 3: Romancing the Star
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResultsPagination Should render correctly if there is a focused movie 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-md-8"
    >
      <div
        className="login card"
      >
        <div
          className="card-header"
        >
          <h4
            className="card-title"
          >
            Kedvencek
          </h4>
        </div>
        <div
          className="card-body"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Reality Bites
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Demolition Man
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Kramer vs. Kramer
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            The Christmas Chronicles: Part Two
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            The Ides of March
          </button>
          <button
            className="btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            The Princess Switch 3: Romancing the Star
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
